delira/utils/config.py

```diff
@@ -1,13 +1,509 @@
+        kwargs:
+            additional keyword arguments passed to `formatter`
+        """
+        with open(path, "r") as f:
+            decoded_format = formatter(f, **kwargs)
+        decoded_format = decoder_cls().decode(decoded_format)
+        if not isinstance(decoded_format, cls):
+            decoded_format = cls(decoded_format)
+        return decoded_format
+
+    @classmethod
+    def loads(cls, data, formatter=yaml.load, decoder_cls=Decoder, **kwargs):
+        """
+        Load config from a string
+
+        Parameters
+        ----------
+        formatter : callable, optional
+            defines the format how the config is saved, by default yaml.dump
+        encoder_cls : :class:`Encoder`, optional
+            trasforms config to a format which can be formatted by the
+            `formatter`, by default Encoder
+        kwargs:
+            additional keyword arguments passed to `formatter`
+        """
+        decoded_format = formatter(data, **kwargs)
+        decoded_format = decoder_cls().decode(decoded_format)
+        if not isinstance(decoded_format, cls):
+            decoded_format = cls(decoded_format)
+        return decoded_format
+
+    @classmethod
+    def create_from_dict(cls, value, deepcopy=False):
+        """
+        Create config from dict like object
+
+        Parameters
+        ----------
+        value : dict like
+            dict like object used to create new config
+        deepcopy : bool, optional
+            if enabled, copies values from origin, by default False
+
+        Returns
+        -------
+        :class:`Config`
+            new config
+        """
+        assert isinstance(value, dict)
+        new_config = cls()
+        for key, item in value.items():
+            if isinstance(item, dict):
+                item = cls(item)
+            if deepcopy:
+                new_config[key] = copy.deepcopy(item)
+            else:
+                new_config[key] = item
+        return new_config
+
+    @classmethod
+    def create_from_argparse(cls, value, deepcopy=False, **kwargs):
+        """
+        Create config from argument parser
+
+        Parameters
+        ----------
+        value : argument parser or namespace
+            if value is an argument parser, the arguments are first parsed
+            and than a new config with the values is created
+            if value is a Namespace the new config is created immediatly
+        deepcopy : bool, optional
+            if enabled, copies values from origin, by default False
+
+        Returns
+        -------
+        :class:`Config`
+            new config
+
+        Raises
+        ------
+        TypeError
+            if value is not an instance of :class:`ArgumentParser`
+            or :class:`Namespace`
+        """
+        if isinstance(value, argparse.ArgumentParser):
+            args_parsed = value.parse_args(**kwargs)
+            return cls.create_from_dict(vars(args_parsed), deepcopy=deepcopy)
```

Comment:
it would be cleaner to call `cls.create_from_argparse(args_parsed. deepcopy=deepcopy)` . If we need to change the `vars` path, we only have to do it in one line, making it easier to maintain. And I think the overhead of an additional function call is negligible

Reply:
No, there isn't, but you could rewrite this to 
```
if isinstance(value, argparse.ArgumentParser):
            args_parsed = value.parse_args(**kwargs)
            return cls.create_from_argparse(args_parsed)
elif isinstance(value, argparse.Namespace):
            return cls.create_from_dict(vars(value), deepcopy=deepcopy)
else:
            raise TypeError("Type of args not supported.")
```
which would be a bit cleaner if we might want to refactor the Namespace part sometime in the future